### PR TITLE
Add REPEAT() example

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

ref #16945

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@dveeden Could you please invite a tech reviewer? Thanks!

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -1226,6 +1226,45 @@ Return the substring that matches the regular expression (Partly compatible with
 
 Repeat a string the specified number of times.
 
+Examples:
+
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:1230]:
```suggestion

The following example generates a sequence of numbers from 1 to 20 using a [recursive common table expression (CTE)](/develop/dev-guide-use-common-table-expression.md#recursive-cte). For each number in the sequence, the character `x` is repeated the number of times equal to the number itself.

```

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -1226,6 +1226,63 @@ Return the substring that matches the regular expression (Partly compatible with
...
+|   16 | xxxxxxxxxxxxxxxx     |
+|   17 | xxxxxxxxxxxxxxxxx    |
+|   18 | xxxxxxxxxxxxxxxxxx   |
+|   19 | xxxxxxxxxxxxxxxxxxx  |
+|   20 | xxxxxxxxxxxxxxxxxxxx |
++------+----------------------+
+20 rows in set (0.01 sec)
+```
+
+The example below demonstrates that `REPEAT()` can handle multi character strings as well.
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:1270]:
```suggestion
The following example demonstrates that `REPEAT()` can operate on operate on strings consisting of multiple characters.
```

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 3]
```diff
@@ -1226,6 +1226,62 @@ Return the substring that matches the regular expression (Partly compatible with
...
+|   16 | xxxxxxxxxxxxxxxx     |
+|   17 | xxxxxxxxxxxxxxxxx    |
+|   18 | xxxxxxxxxxxxxxxxxx   |
+|   19 | xxxxxxxxxxxxxxxxxxx  |
+|   20 | xxxxxxxxxxxxxxxxxxxx |
++------+----------------------+
+20 rows in set (0.01 sec)
+```
+
+The following example demonstrates that `REPEAT()` can operate on operate on strings consisting of multiple characters.
```
Comment (Oreoxmt) [inline functions-and-operators/string-functions.md:1270]:
```suggestion
The following example demonstrates that `REPEAT()` can operate on strings consisting of multiple characters.
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**